File: dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI/AI/Prompt/PromptResponse.cs
using Microsoft.Bot.Schema;
using Microsoft.Teams.AI.AI.Models;

namespace Microsoft.Teams.AI.AI.Prompt
{
    /// <summary>
    /// Response returned by a `PromptCompletionClient`.
    /// </summary>
    public class PromptResponse
    {
        /// <summary>
        /// Status of the prompt response.
        /// </summary>
        public PromptResponseStatus Status { get; set; }

        /// <summary>
        /// Message returned.
        /// </summary>
        public ChatMessage? Message { get; set; }

        /// <summary>
        /// Error returned.
        /// </summary>
        public Error? Error { get; set; }
    }

    /// <summary>
    /// Status of the prompt response.
    /// </summary>
    public enum PromptResponseStatus
    {
        /// <summary>
        /// Success - The prompt was successfully completed.
        /// </summary>
        Success,
        /// <summary>
        /// Error - An error occurred while completing the prompt.
        /// </summary>
        Error,
        /// <summary>
        /// RateLimited - The request was rate limited.
        /// </summary>
        RateLimited,
        /// <summary>
        /// InvalidResponse - The response was invalid.
        /// </summary>
        InvalidResponse,
        /// <summary>
        /// TooLong - The rendered prompt exceeded the `max_input_tokens` limit.
        /// </summary>
        TooLong
    }
}

File: python/packages/ai/teams/ai/models/prompt_response.py
"""
Copyright (c) Microsoft Corporation. All rights reserved.
Licensed under the MIT License.
"""

from __future__ import annotations

from dataclasses import dataclass
from typing import Any, Generic, List, Literal, Optional, TypeVar, Union

from ..prompts.message import Message

ContentT = TypeVar("ContentT")
PromptResponseStatus = Literal["success", "error", "rate_limited", "invalid_response", "too_long"]


@dataclass
class PromptResponse(Generic[ContentT]):
    """
    Response returned by a `PromptCompletionClient`.
    """

    status: PromptResponseStatus = "success"
    """
    Status of the prompt response.
    """

    input: Optional[Union[Message[Any], List[Message[Any]]]] = None
    """
    Input message sent to the model. `undefined` if no input was sent.
    """

    message: Optional[Message[ContentT]] = None
    """
    Message returned.
    """

    error: Optional[str] = None
    """
    Error returned.
    """
